### PR TITLE
Add a request access flow to the unauthorized error screen

### PR DESCRIPTION
**Pull request by jaril**
This adds the following:
- Show a request access button when a logged-in user is viewing a replay they're not authorized to.
- Show a list of collaborator requests in the sharing modal.

Let's add some docs (@ceceliacreates) here if we need to. Definitely get some qa wolf tests.

Demo: https://share.descript.com/view/r18vVVcnm4z

https://user-images.githubusercontent.com/15959269/151447826-a2d290c5-3504-4ffa-840f-c9114e9e0691.mov

### CODE REVIEW

File: src/ui/components/shared/Error.tsx [hunk 1]
```diff
@@ -103,6 +130,8 @@ function ActionButton({ action }: { action: string }) {
     button = <LibraryButton />;
   } else if (action == "team-billing") {
     button = <TeamBillingButton />;
+  } else if (action == "request-access") {
```
Comment (abstractalgo) [inline src/ui/components/shared/Error.tsx:133]: `===`

**Review by jasonLaster — APPROVED** [review]

**Comment by jcmorrow** [comment]
This is maybe a silly request - but do you think it would be worthwhile to put min heights/widths on the dialog so that when it changes from one message to another it does not change shape slightly?

**Comment by jaril** [comment]
> This is maybe a silly request - but do you think it would be worthwhile to put min heights/widths on the dialog so that when it changes from one message to another it does not change shape slightly?

I'm down. I think @jonbell-lot23 is in favor of things not shifting around too so should be good.

File: src/ui/hooks/recordings.ts [hunk 2]
```diff
@@ -114,6 +114,8 @@ export async function getRecording(recordingId: RecordingId) {
     query: GET_RECORDING,
     variables: { recordingId },
   });
+
+  console.log({ data: result.data });
```
Comment (jasonLaster) [inline src/ui/hooks/recordings.ts:118]: 🧂

File: src/ui/hooks/recordings.ts [hunk 3]
```diff
@@ -802,3 +806,30 @@ export function useRequestRecordingAccess() {
...
+
+  if (data) {
+    console.log({ data });
+  }
+  if (loading) {
+    console.log({ loading });
+  }
+  if (error) {
+    console.log({ error });
+  }
```
Comment (jasonLaster) [inline src/ui/hooks/recordings.ts:832]: 🧂

File: src/ui/components/Library/ViewerRouter.tsx [hunk 4]
```diff
@@ -83,6 +83,8 @@ function ViewerRouter(props: ViewerRouterProps) {
   const { features, loading } = hooks.useGetUserInfo();
   const { currentWorkspaceId } = props;
 
+  console.log({ currentWorkspaceId });
```
Comment (jasonLaster) [inline src/ui/components/Library/ViewerRouter.tsx:86]: 🧂

**Review by jasonLaster — APPROVED** [review]
Code looks legit.

**Comment by ceceliacreates** [comment]
Yes. This would work well in the Library page and also the upcoming Replay overview/What is Replay section.

**Pull request merged**